pkg/controller/operator/seed/resources/metering/deployment.go

```diff
@@ -0,0 +1,323 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metering
+
+import (
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	meteringToolName       = "kubermatic-metering"
+	meteringDataName       = "metering-data"
+	meteringCronJobMonthly = "kubermatic-metering-report-monthly"
+)
+
+// DeploymentCreator creates a new metering tool deployment per seed cluster.
+func DeploymentCreator(_ *kubermaticv1.Seed) reconciling.NamedDeploymentCreatorGetter {
+	return func() (string, reconciling.DeploymentCreator) {
+		//TODO: Add custom values for the metering deployment fields such as seed, interval and output-rotation.
+		return meteringToolName, func(d *appsv1.Deployment) (*appsv1.Deployment, error) {
+			d.Spec.Replicas = pointer.Int32Ptr(1)
+			d.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"role":    meteringToolName,
+					"version": "1",
+				},
+			}
+
+			d.Spec.Strategy.Type = appsv1.RecreateDeploymentStrategyType
+
+			d.Spec.Template.Labels = map[string]string{
+				"role":    meteringToolName,
+				"version": "1",
+			}
+			d.Spec.Template.Annotations = map[string]string{
+				"kubermatic/scrape":      "true",
+				"kubermatic/scrape_port": "2112",
```

Comment:
Please use `prometheus.io/scrape` and `prometheus.io/port`

Reply:
This is was mainly inspired by those templates so I just did a copy and paste https://github.com/kubermatic/metering/blob/master/config/kubermatic-metering/templates/kubermatic-metering-dep.yaml#L19-L20. I can check why they were left as Kubermatic instead of Prometheus.

Reply:
Those metrics are not exported anymore, so you can omit it for now.

Reply:
@WeirdMachine Why those metrics are not exported anymore?

Reply:
I would like to define first what kind of metric we export and why. The initial exposed data seems not relevant to me.

Reply:
Yeah that's no problem, I think we can use the Prometheus tags for now and once you decide what things need to be adjusted we can simply adjust them.